views/mdc/assets/js/components/events/autocomplete.js

```diff
@@ -0,0 +1,94 @@
+import {VBase} from './base';
+
+// Auto complete a datalist
+// The elementId is a datalist
+// The url is called as a GET expecting json back
+export class VAutoComplete extends VBase {
+    constructor(options, url, params, event) {
+        super(options);
+        this.element_id = options.target;
+        this.url = url;
+        this.params = params;
+        this.event = event;
+    }
+
+    call(results) {
+        // Clear the timeout if it has already been set.
+        // This will prevent the previous task from executing
+        // if it has been less than <MILLISECONDS>
+        let parentElement = this.parentElement();
+        let updateElement = this.createUpdateElementHandler(this);
+        let promiseObj = new Promise(function (resolve) {
+            clearTimeout(parentElement.vTimeout);
+            // Make a new timeout
+            parentElement.vTimeout = setTimeout(updateElement, 500);
+            results.push({action:'autocomplete', statusCode: 200});
+            resolve(results);
+        });
+        return promiseObj;
+    }
+
+    // This is used to get a proper binding of the object
+    // https://stackoverflow.com/questions/750486/javascript-closure-inside-loops-simple-practical-example
+    createUpdateElementHandler(_this_) {
+        return function () {
+            _this_.updateElement();
+        };
+    }
+
+    updateElement() {
+        this.clearErrors();
+        this.getData(this.populateOptions);
+    }
+
+    dataList() {
+        return document.getElementById(this.element_id);
```

Comment:
To be shadow dom ready, use the root we have
```suggestion
        return this.root.getElementById(this.element_id);
```